mypy/semanal.py

```diff
@@ -2159,6 +2160,51 @@ def build_typeddict_typeinfo(self, name: str, items: List[str],
 
         return info
 
+    def check_classvar(self, s: AssignmentStmt) -> None:
+        lvalue = s.lvalues[0]
+        if len(s.lvalues) != 1 or not isinstance(lvalue, NameExpr):
+            return
+        is_classvar = self.check_classvar_definition(lvalue, s.type)
+        if self.is_class_scope() and isinstance(lvalue.node, Var):
+            # Assignments to class variables outside class scope are checked later
+            self.check_classvar_override(lvalue.node, is_classvar)
+
+    def check_classvar_definition(self, lvalue: NameExpr, typ: Type) -> bool:
+        if not isinstance(typ, UnboundType):
+            return False
+        sym = self.lookup_qualified(typ.name, typ)
+        if not sym or not sym.node:
+            return False
+        fullname = sym.node.fullname()
+        if fullname != 'typing.ClassVar':
+            return False
+        if self.is_class_scope() or not isinstance(lvalue.node, Var):
+            node = cast(Var, lvalue.node)
+            node.is_classvar = True
```

Comment:
Indeed, that was wrong. I had if statements written a bit different at first and didn't fully apply De Morgan's law there. And because mypy was complaining there (and I didn't see anything wrong in it...), I've added cast.